trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/SigningController.java

```diff
@@ -36,17 +37,56 @@ public SigningController(CredentialsController credentialsController, SigningCon
         maxClockDrift = signingControllerConfig.getMaxClockDrift().toJavaTime();
     }
 
+    public Optional<SigningMetadata> signingMetadataFromRequest(
+            Function<Credentials, Credentials.Credential> credentialsSupplier,
```

Comment:
I understand that this is to support us obtaining the metadata in the same way whether this is using the emulated access key or the external one. Is the idea that we may want to proxy requests signed with the real access key?

However, we currently never use this argument. Even if we were to use it, we would need to somehow pass it onto `isValidAuthorization` and we should have a way to retrieve a Credentials pair from a real key in the same way we do for emulated ones right?

Reply:
> Is the idea that we may want to proxy requests signed with the real access key?

We need to sign the real request being sent to AWS. The signing mechanism is the same. However, for the real request we must use the real AccessKey/Secret. 

> However, we currently never use this argument

It's used by `TrinoS3ProxyClient`. See here: https://github.com/trinodb/s3-proxy/pull/11/files#diff-86567ef27fb9d1f3e89e0d28033e383903d06ae883810cb03d45ca40400bb665R104

Reply:
It is used for the `sign` method but not for the `signingMetadataFromRequest` right? I completely agree with needing this for the `sign` method, just wondering its purpose for this one

Reply:
I see. `signingMetadataFromRequest` always uses the emulated credentials. I find it easier to reason about when `SigningController` is agnostic about which credentials are used. I'd prefer to leave as is unless you feel strongly about it.

Reply:
`SigningController` can be agnostic as to what credentials are used, I agree that's good. But this method is taking an argument that is unused, so it is misleading for a caller as it gives the impression you can ask it to use the real credentials, but it won't.

If we want to make it explicit that this method (unlike the others in `SigningController`) only deals with emulated credentials, should we rename to `validateEmulatedCredentialsAndGetMetadata`?

Reply:
Hold on - I'm an idiot. I see the problem now. That argument _should_ be used. 1 second I'll fix it.

Reply:
OK - it's fixed now. PTAL. Thanks for your persistence.

Reply:
No worries! I originally had this design in mind, but at present there is no way to allow users to pass in `Credentials::real` to this method (as our credential store doesn't store a mapping from real credential to emulated ones, only the other way around)

However, that is personal preference - and in any case, we could tweak the CredentialsController later if we wanted to make this possible for any reason.

Looks good!

Reply:
Yes ^^^ - we should leave it due to that.